component-monitor/src/main.rs

```diff
@@ -0,0 +1,238 @@
+            Ok(()) => {
+                debug!(target: "files", "removed sentinel ({0:?})", &sentinel_path)
+            }
+            Err(e) => {
+                warn!(target: "files", "failed to remove {0:?}: {e}", &sentinel_path)
+            }
+        }
+    }
+    for entry in fs::read_dir(target).expect("ERROR: failed to list target directory") {
+        match entry {
+            Ok(e) => {
+                let path = e.path();
+                if path.is_dir() {
+                    match fs::remove_dir_all(&path) {
+                        Ok(()) => {
+                            debug!(target: "files", "removed {0:?} recursively", &path)
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to remove {0:?}: {e}", &path)
+                        }
+                    }
+                } else {
+                    match fs::remove_file(&path) {
+                        Ok(()) => {
+                            debug!(target: "files", "removed {0:?}", &path)
+                        }
+                        Err(e) => {
+                            error!(target: "files", "failed to remove {0:?}: {e}", &path)
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                error!(target: "files", "{e}");
+            }
+        }
+    }
+}
+
+/// Check if the sentinel file is current, else clean the target directory and populate again,
+/// sentinel file being last
+fn populate(source: &PathBuf, sentinel: &String, target: &PathBuf) {
+    info!(target: "files", "populating {target:?} from {source:?} with sentinel {sentinel:?}");
+    let sentinel_src = source.join(sentinel);
+
+    if !sentinel_src.exists() {
+        info!(target: "files", "sentinel file ({0:?}) not found, skipping", &sentinel_src);
+        return;
+    }
+
+    let sentinel_data = match fs::read(&sentinel_src) {
+        Ok(d) if d.trim_ascii() == b"" => {
+            error!(target: "files", "found empty sentinel file ({0:?})", &sentinel_src);
+            return;
+        }
+        Ok(d) => d,
+        Err(e) => {
+            error!(target: "files", "failed to read sentinel file: {e}");
+            return;
+        }
+    };
+
+    let sentinel_tgt = target.join(sentinel);
+    if sentinel_data == fs::read(&sentinel_tgt).unwrap_or_default() {
+        info!(target: "files", "found current sentinel, skipping");
+        return;
+    }
+
+    cleanup(target, sentinel);
+    info!(target: "files", "copying from {source:?} to {target:?}");
+    for entry in WalkDir::new(&source) {
+        match entry {
+            Ok(e) => {
+                let path = e.path();
+                let relative_path = path.strip_prefix(source).unwrap();
+                if path == source {
+                    continue;
+                }
+                if *(relative_path.to_str().unwrap()) == **sentinel {
```

Comment:
There should be a way to get this to read `relative_path == sentinel.into()` and then compare them as `PathBuf`